### PR TITLE
ramips: use more devm in mtk_eth_soc

### PR DESCRIPTION
**Pull request by neheb**
ping @DragonBluep @981213

### CODE REVIEW

**Comment by neheb** [comment]
That devm_iounmap looks suspicious.

**Comment by neheb** [comment]
Yeah I think it's a mistake.

**Comment by Vladdrako** [comment]
Maybe it's my local issue, but...
mt7620a/asus_ac51u
```
drivers/net/ethernet/ralink/mtk_eth_soc.c: In function 'fe_probe':
drivers/net/ethernet/ralink/mtk_eth_soc.c:1544:50: error: passing argument 1 of 'devm_platform_ioremap_resource' from incompatible pointer type [-Wincompatible-pointer-types]
 1544 |         fe_base = devm_platform_ioremap_resource(&pdev->dev, 0);
      |                                                  ^~~~~~~~~~
      |                                                  |
      |                                                  struct device *
In file included from drivers/net/ethernet/ralink/mtk_eth_soc.c:23:
./include/linux/platform_device.h:72:56: note: expected 'struct platform_device *' but argument is of type 'struct device *'
   72 | devm_platform_ioremap_resource(struct platform_device *pdev,
      |                                ~~~~~~~~~~~~~~~~~~~~~~~~^~~~

```

**Comment by neheb** [comment]
Yeah. Mistake. Should be pdev

**Comment by neheb** [comment]
ping @robimarko

**Comment by namiltd** [comment]
The commit causes repeated errors on TP-Link Archer C5-v4 (ramips/mt7620) causing LAN/WAN not to work:
```
Fri Sep 27 20:21:19 2024 kern.crit kernel: [  272.706689] mtk_soc_eth 10100000.ethernet eth0: NETDEV WATCHDOG: CPU: 0: transmit queue 1 timed out 12080 ms
Fri Sep 27 20:21:19 2024 kern.err kernel: [  272.716785] mtk_soc_eth 10100000.ethernet eth0: transmit timed out
Fri Sep 27 20:21:19 2024 kern.info kernel: [  272.723097] mtk_soc_eth 10100000.ethernet eth0: dma_cfg:80000065
Fri Sep 27 20:21:19 2024 kern.info kernel: [  272.729262] mtk_soc_eth 10100000.ethernet eth0: tx_ring=0, base=02ce8000, max=1024, ctx=0, dtx=0, fdx=0, next=0
Fri Sep 27 20:21:19 2024 kern.info kernel: [  272.739586] mtk_soc_eth 10100000.ethernet eth0: rx_ring=0, base=02ce4000, max=1024, calc=1023, drx=0
Fri Sep 27 20:21:19 2024 kern.info kernel: [  272.750243] mtk_soc_eth 10100000.ethernet eth0: port 5 link down
Fri Sep 27 20:21:19 2024 kern.info kernel: [  272.758104] mtk_soc_eth 10100000.ethernet eth0: port 5 link up (1000Mbps/Full duplex)
Fri Sep 27 20:21:24 2024 kern.crit kernel: [  275.596732] mtk_soc_eth 10100000.ethernet eth0: NETDEV WATCHDOG: CPU: 0: transmit queue 1 timed out 17040 ms
Fri Sep 27 20:21:24 2024 kern.err kernel: [  275.606835] mtk_soc_eth 10100000.ethernet eth0: transmit timed out
Fri Sep 27 20:21:24 2024 kern.info kernel: [  275.613154] mtk_soc_eth 10100000.ethernet eth0: dma_cfg:80000065
Fri Sep 27 20:21:24 2024 kern.info kernel: [  275.619321] mtk_soc_eth 10100000.ethernet eth0: tx_ring=0, base=02ce8000, max=1024, ctx=0, dtx=0, fdx=0, next=0
Fri Sep 27 20:21:24 2024 kern.info kernel: [  275.629645] mtk_soc_eth 10100000.ethernet eth0: rx_ring=0, base=02cec000, max=1024, calc=1023, drx=0
Fri Sep 27 20:21:24 2024 kern.info kernel: [  275.640258] mtk_soc_eth 10100000.ethernet eth0: port 5 link down
Fri Sep 27 20:21:24 2024 kern.info kernel: [  275.648131] mtk_soc_eth 10100000.ethernet eth0: port 5 link up (1000Mbps/Full duplex)
```
Reverting the commit fixes this issue.

File: target/linux/ramips/files/drivers/net/ethernet/ralink/mtk_eth_soc.c [hunk 1]
```diff
@@ -1542,17 +1541,14 @@ static int fe_probe(struct platform_device *pdev)
...
-	if (IS_ERR(fe_base)) {
-		err = -EADDRNOTAVAIL;
-		goto err_out;
-	}
+	fe_base = devm_platform_ioremap_resource(pdev, 0);
+	if (IS_ERR(fe_base))
+		return PTR_ERR(fe_base);
 
-	netdev = alloc_etherdev(sizeof(*priv));
+	netdev = devm_alloc_etherdev_mqs(&pdev->dev, sizeof(*priv), 4, 4);
```
Comment (981213) [inline target/linux/ramips/files/drivers/net/ethernet/ralink/mtk_eth_soc.c:1548]: This old ralink driver doesn't use multiple queue I think?

Comment (neheb) [reply target/linux/ramips/files/drivers/net/ethernet/ralink/mtk_eth_soc.c:1548]: I wonder where I got this 4 from...

@namiltd try

```
--- a/target/linux/ramips/files/drivers/net/ethernet/ralink/mtk_eth_soc.c
+++ b/target/linux/ramips/files/drivers/net/ethernet/ralink/mtk_eth_soc.c
@@ -1545,7 +1545,7 @@ static int fe_probe(struct platform_device *pdev)
        if (IS_ERR(fe_base))
                return PTR_ERR(fe_base);
 
-       netdev = devm_alloc_etherdev_mqs(&pdev->dev, sizeof(*priv), 4, 4);
+       netdev = devm_alloc_etherdev(&pdev->dev, sizeof(*priv));
        if (!netdev) {
                dev_err(&pdev->dev, "alloc_etherdev failed\n");
                return -ENOMEM;

```

Comment (namiltd) [reply target/linux/ramips/files/drivers/net/ethernet/ralink/mtk_eth_soc.c:1548]: It works, please prepare PR

**Pull request merged**